metabase/metabase
# Navbar Improvements: Updated Heuristics

### Description

Implements new logic for the navbar being open/closed for a page and auto-closing in certain conditions.

This is the core change to complete https://github.com/metabase/metabase/issues/36370

### Technical description

The main part of this change is having the navbar state subscribe to react-router-redux `@@router/LOCATION_CHANGE`  event to decide if we should keep the navbar open/closed on route changes. A special case is accommodated for home page redirects where we want to treat a url we would typically auto-close for as if it were the normal home page. This is accomplished by adding a flag to the location state on the replace event which is used to return the prev state / ignore the event.


### How to verify

- Log into metabase, should see navbar open
- Visit a dashboard, should see navbar auto-close
- Click the MB logo, should stay closed
- Go back a page
- Manually open the navbar
- Click the MB logo, should stay open
- Go to a collection, should stay open
- Go to a dashboard within a collection, should auto-close
- Using the breadcrumbs at the top of the dashboard, go back to the collection, should stay closed
- Manually open the navbar
- Create a new question or sql query, should auto close


### Demo

https://github.com/metabase/metabase/assets/7104357/fc7c30c2-abe5-4504-b150-1ba083e23d71
(demo hangs at the end due to a perf issue unrelated to this change)

### Checklist
